File: RepoTransBench/target_projects/Java/Go/yandex-qatools_postgresql-embedded/public_tests/cached_postgres_artifact_store_public_test.go
package public_tests

import (
	"strings"
	"testing"
)

type DummyCache struct {
	fileName     string
	timesFetched int
}

func NewDummyCache(fileName string) *DummyCache {
	return &DummyCache{fileName: fileName}
}

func (c *DummyCache) Fetch() string {
	c.timesFetched++
	return "fetched_public_" + c.fileName
}

func (c *DummyCache) GetTimesFetched() int {
	return c.timesFetched
}

func TestFetchReturnsFileWithDifferentName(t *testing.T) {
	cache := NewDummyCache("pubfile-2211.txt")
	res := cache.Fetch()
	if !strings.Contains(res, "pubfile-2211.txt") {
		t.Errorf("Expected fetch result with pubfile-2211.txt, got '%s'", res)
	}
}

func TestFetchCountsTimesFetchedWithDifferentFile(t *testing.T) {
	cache := NewDummyCache("pubcache.data")
	cache.Fetch()
	cache.Fetch()
	if cache.GetTimesFetched() != 2 {
		t.Errorf("Expected timesFetched to be 2, got %d", cache.GetTimesFetched())
	}
}

File: RepoTransBench/source_projects/Java/yandex-qatools_postgresql-embedded/src/test/java/de/flapdoodle/embed/process/store/CachedPostgresArtifactStorePublicTest.java
package de.flapdoodle.embed.process.store;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Public test for CachedPostgresArtifactStore using different values.
 * Dummy cache used for compilation (remove if real cache class exists).
 */
public class CachedPostgresArtifactStorePublicTest {

    static class DummyCache {
        private String fileName;
        private int timesFetched = 0;

        public DummyCache(String fileName) {
            this.fileName = fileName;
        }

        public String fetch() {
            timesFetched++;
            return "fetched_public_" + fileName;
        }

        public int getTimesFetched() {
            return timesFetched;
        }
    }

    @Test
    public void fetchReturnsFileWithDifferentName() {
        DummyCache cache = new DummyCache("pubfile-2211.txt");
        String res = cache.fetch();
        assertTrue(res.contains("pubfile-2211.txt"));
    }

    @Test
    public void fetchCountsTimesFetchedWithDifferentFile() {
        DummyCache cache = new DummyCache("pubcache.data");
        cache.fetch();
        cache.fetch();
        assertEquals(2, cache.getTimesFetched());
    }
}